aircmd/actions/environments.py

```diff
@@ -234,24 +235,24 @@ def with_bound_docker_host(
 
 
 def with_docker_cli(
-    context: PipelineContext, settings: GlobalSettings, shared_volume: Optional[Tuple[str, CacheVolume]] = None, docker_service_name: Optional[str] = None
+    pipeline: Pipeline, settings: GlobalSettings, shared_volume: Optional[Tuple[str, CacheVolume]] = None, docker_service_name: Optional[str] = None
```

Comment:
Is globalsettings considered the way I pass configuration to environments?

If so what if two sibling pipelines use the same `with_*` but they differ in the settings they apply

Reply:
Settings right now is intended to be a global, immutable singleton so the inputs would be identical. I should codify that by making it frozen/immutable and subclassing singleton

Reply:
I just want to clarify that when I say "Identical", i mean within the context of a single pipeline run. One interesting implication here is that I don't think that would be the case across multiple runs unless the configuration is obviously identical. It might make sense to not actually pass config in because of that if we cared about layer caching across runs. I will say that the existing PipelineContext pattern that I lifted this from also included configuration info, so it's not really a regression from the current code that runs this in airbytehq/airbyte, but we could probably make it better in this way by not passing settings in. I'm not sure if that will work yet though.